spec/api-protocol-spec.js

```diff
@@ -897,6 +912,21 @@ describe('protocol module', () => {
         })
       })
     })
+
+    it('can access headers', (done) => {
+      const handler = (request, callback) => {
+        assert.ok('headers' in request)
+        done()
+      }
+      protocol.interceptHttpProtocol('http', handler, (error) => {
```

Comment:
Please assume that `protocol.interceptHttpProtocol()` works here and don't handle any possible errors.
A test should tests exactly that it supposed to test and nothing else.

```js
const handler = (request) => {
  assert.ok('headers' in request)
  done()
}
protocol.interceptHttpProtocol('http', handler, () => {
  fetch('http://fake-host')
  // `handler` will intercept the request.
})
```

Reply:
You're right, it wasn't clear to me at the time I wrote the test that a timeout would occur if `done()` was never invoked, causing the test to fail anyway. I'll make this change.

However, I'd like to point out that this is frustrating when contributing to open source projects like Electron. Projects have style guides and conventions that a casual contributor can't reasonably learn all of before providing a small PR. Matching existing style in a file is important to prevent it from turning into a mess of different styles. I emulated the test directly above this one, and through the file `if (error) return done(error)` is used 44 times, most of them in cases like this where it's not actually necessary. So after making this change you've requested (which is valid and I think a good change), this test will have a different style than the rest. I can't read minds, half the time in a project a PR will require changes for *not* following the existing style, half the time it will require changes for not being as tight as it could be (as in this case).

Reply:
@dsanders11 Thank you for the feedback.
I totally agree with you and have to apologise that some Electron code is not good as it could be. We are constantly working on improving both the app code and its tests, but it is continuous process and is far away from being finished.